c10/cuda/CUDAStream.cpp

```diff
@@ -369,6 +378,18 @@ void setCurrentCUDAStream(CUDAStream stream) {
   current_streams[ptr->device_index] = ptr;
 }
 
+CUDAStream registerCustomCUDAStream(DeviceIndex device_index, CUDAStreamProvider stream_provider) {
```

Comment:
why do you need stream_provider? can't you just pass cudaStream_t directly?

Reply:
This is to allow hooks. I have RAII style hooks in mind. That said, I haven't tested this would work. (Haven't written any C++ for the past 5 years 😢)